From 074873c68e05c84c15aace2fbe02cea15aaf767c Mon Sep 17 00:00:00 2001
From: Shengjing Zhu <zhsj@debian.org>
Date: Sat, 23 Jan 2021 01:26:42 +0800
Subject: [PATCH] Add cgo tag to btrfs plugin

btrfs plugin needs CGO support. However on riscv64, cgo
is only support on go1.16 (not released yet).
Instead of setting no_btrfs manually, adding a cgo tag tells
the compiler to skip it automatically.

Signed-off-by: Shengjing Zhu <zhsj@debian.org>
---
 cmd/containerd/builtins_btrfs_linux.go | 2 +-
 snapshots/btrfs/btrfs.go               | 2 +-
 snapshots/btrfs/btrfs_test.go          | 2 +-
 snapshots/btrfs/plugin/plugin.go       | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/cmd/containerd/builtins_btrfs_linux.go b/cmd/containerd/builtins_btrfs_linux.go
index 5a0dd7a51e..7eb7095249 100644
--- a/cmd/containerd/builtins_btrfs_linux.go
+++ b/cmd/containerd/builtins_btrfs_linux.go
@@ -1,4 +1,4 @@
-// +build !no_btrfs
+// +build !no_btrfs,cgo
 
 /*
    Copyright The containerd Authors.
diff --git a/snapshots/btrfs/btrfs.go b/snapshots/btrfs/btrfs.go
index 27b734af29..3096989aee 100644
--- a/snapshots/btrfs/btrfs.go
+++ b/snapshots/btrfs/btrfs.go
@@ -1,4 +1,4 @@
-// +build linux,!no_btrfs
+// +build linux,!no_btrfs,cgo
 
 /*
    Copyright The containerd Authors.
diff --git a/snapshots/btrfs/btrfs_test.go b/snapshots/btrfs/btrfs_test.go
index ceff97a261..03789407ec 100644
--- a/snapshots/btrfs/btrfs_test.go
+++ b/snapshots/btrfs/btrfs_test.go
@@ -1,4 +1,4 @@
-// +build linux,!no_btrfs
+// +build linux,!no_btrfs,cgo
 
 /*
    Copyright The containerd Authors.
diff --git a/snapshots/btrfs/plugin/plugin.go b/snapshots/btrfs/plugin/plugin.go
index 4f4f854087..a70af58429 100644
--- a/snapshots/btrfs/plugin/plugin.go
+++ b/snapshots/btrfs/plugin/plugin.go
@@ -1,4 +1,4 @@
-// +build linux,!no_btrfs
+// +build linux,!no_btrfs,cgo
 
 /*
    Copyright The containerd Authors.
